### ISSUE TITLE
DOCKER

### ISSUE THREAD

**Issue by wahmedswl** [logicahealth/InfoButtons#15]
Hi, it would be great to have DOCKER Image available with Docker Compose for full spinning up container. As, it will help spin it up Quickly.

Thanks

**Comment by aniskand** [comment]
Hi, 

At the moment we don't have the development bandwidth for both Docker and Vagrant, so we're sticking with Vagrant for now. Thanks.

**Comment by wahmedswl** [comment]
@aniskand thanks for your feedback. Would it be possible to just give full instruction to run the InfoButton locally?

As, i have local build of InfoButton which has been generated according to the BUILDGUIDE, but i guess i am missing some configuration for DB as when running the 

http://localhost:8080/infobutton-service/infoRequest?representedOrganization.id.root=1.3.6.1.4.1.3768&patientPerson.administrativeGenderCode.c=F&age.v.v=47&age.v.u=a&taskContext.c.c=PROBLISTREV&mainSearchCriteria.v.c=44054006&mainSearchCriteria.v.cs=2.16.840.1.113883.6.96&mainSearchCriteria.v.dn=Diabetes%20mellitus%20type%202&performer=PROV&informationRecipient=PROV

```could not execute statement```

I will try to create DOCKER image if i be able to run it Locally. Also, i created ticket about DB Scripts, see #11, it would be great if Scripts are ordered or any reference where order of their execution can be obtained.

Thanks

**Comment by aniskand** [comment]
Could you give me a bit more detail as to the error you're getting? If you can show me the output from the log in catalina.out that would be more helpful. Also, which branch of OpenInfobutton did you build? Thanks.

**Comment by wahmedswl** [comment]
@aniskand i build from latest ```development``` branch. PFA

[catalina.2017-05-31.txt](https://github.com/VHAINNOVATIONS/InfoButtons/files/1040828/catalina.2017-05-31.txt)
[catalina.2017-05-29.txt](https://github.com/VHAINNOVATIONS/InfoButtons/files/1040824/catalina.2017-05-29.txt)
[catalina.2017-05-27.txt](https://github.com/VHAINNOVATIONS/InfoButtons/files/1040825/catalina.2017-05-27.txt)
[catalina.2017-05-18.txt](https://github.com/VHAINNOVATIONS/InfoButtons/files/1040826/catalina.2017-05-18.txt)
[catalina.2017-05-16.txt](https://github.com/VHAINNOVATIONS/InfoButtons/files/1040827/catalina.2017-05-16.txt)
[catalina.2017-05-09.txt](https://github.com/VHAINNOVATIONS/InfoButtons/files/1040829/catalina.2017-05-09.txt)

Thanks

**Comment by jxfish2** [comment]
> 
> 
> Hi,
> 
> At the moment we don't have the development bandwidth for both Docker and Vagrant, so we're sticking with Vagrant for now. Thanks.

Hi Andrew,

I understand that you either want, or like working with Vagrant.

But, you must understand that there are lots of issues with Vagrant, due to the various different platforms chosen by various different enterprises.

Docker is the logical choice, across the board, for anything that can be containerized today.

If you move away from Vagrant, you will not have so many issues.

If you move to Docker, it will make life so much easier for all of us.

Believe me, I appreciate the effort you've put into this, but Docker automatically makes all of your development efforts cross-platform compatible.

Docker makes deploying your programs as simple as 1, 2, 3...

I truly hope that you will abandon Vagrant, and port to Docker instead.

The industry standard for enterprise environments is RedHat.

Developing in Ubuntu immediately causes each of us in standardized enterprise environments, to edit, tweak and debug, in our RedHat enterprise environments.

Developing in Docker containers, and it does not matter the underlying operating system.

Just my 2 cents worth.

**Comment by aniskand** [comment]
Hi Joe,

As I said in the other thread, the Vagrant Infobutton VM is no longer being developed. Going forward, we are working on a Docker deployment. Thanks.

**Comment by DSSUSER** [comment]
Hi Andrew,
In a private exchange, I requested that you create a complete docker image, without any networking information in it.
Then place this image in a public repository.
We only need to know which ports the image itself is configured to listen on.
It is much easier to run a pre-built image, than to work with Dockerfiles.
Some databases, and some applications, will not work properly when attempting to start, using service or systemd commands, because Docker images do not have their own kernel, or access to the underlying host OS kernel.
But, if you have an image that's working, it would be a piece of cake for any of your followers to simply run that image from a public repository.
I will be happy to help, by providing full instructions on how to download and start the image, with full network functionality.

**Comment by aniskand** [comment]
Yes, I'm working on an update that will automate the deployment of the database as well. It should not require any ip configuration.

**Comment by DSSUSER** [comment]
Thank you so very much.
Have a wonderful day.

**Comment by DSSUSER** [comment]
GM Andrew,
Any updates on a published OIB image?

**Comment by DSSUSER** [comment]
GM Andrew,

Do you have a link to your latest Docker InfoButtons image?

**Comment by gdelfiol** [comment]
Docker is at  https://hub.docker.com/r/aniskand2/infobutton
